src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -46,13 +46,11 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 		handleSubmit: reactFormHookSubmit,
 		getValues,
 		setError,
-		control,
-		formState: { errors },
+		formState,
 		clearErrors,
 	} = formMethods;
 
-	const formValues = useWatch({ control, disabled: !errors });
-	const oldFormValues = usePrevious(formValues);
+	const [oldFormValues, setOldFormValues] = useState<TFrontendEngineValues>({});
```

Comment:
ideally a `useRef` will be better because it will never trigger a rerender. however it is still fine now because the state update happens on a formState update so it is already rerendering.